### PR TITLE
feat: Configurable live seekable window

### PR DESCRIPTION
**Pull request by dario-fiore**
## Description
During live streaming in low bandwidth situations `playback-watcher` try different times to seek to live edge without waiting for buffered contents. This behaviour causing waiting status "loop" because buffer will never reach minimum contents in order to perform playback.

## Specific Changes proposed
Configure method `beforeSeekableWindow_` in order to accept new configuration parameter `liveRangeSafeTimeDelta` to adjust safe time delta.

## Requirements Checklist
- [X] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [X] Unit Tests updated or fixed
  - [X] Docs/guides updated
  - [X] Example created ([JSBin](https://jsbin.com/vuguqik/edit?html,output))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: src/playback-watcher.js [hunk 1]
```diff
@@ -502,7 +503,7 @@ export default class PlaybackWatcher {
     if (seekable.length &&
         // can't fall before 0 and 0 seekable start identifies VOD stream
         seekable.start(0) > 0 &&
-        currentTime < seekable.start(0) - Ranges.SAFE_TIME_DELTA) {
+        currentTime < seekable.start(0) - (this.liveRangeSafeTimeDelta || Ranges.SAFE_TIME_DELTA)) {
```
Comment (gkatsev) [inline src/playback-watcher.js:506]: What about using `this.liveRangeSafeTimeDelta` here always and then in https://github.com/videojs/http-streaming/blob/c9b195b8d8451118960aba2977c1bbc95fda0120/src/videojs-http-streaming.js#L639-L643 defaulting it to `Ranges.SAFE_TIME_DELTA` if a value wasn't provided. Probably by adding an object before `this.options` there.
```js
videojs.mergeOptions({
  liveRangeSafeTimeDelta: Ranges.SAFE_TIME_DELTA
},
this.options_,
{
  seekable: () => this.seekable(),
  media: () => this.masterPlaylistController_.media(),
  masterPlaylistController: this.masterPlaylistController_
});
```

File: README.md [hunk 2]
```diff
@@ -451,6 +452,11 @@ This option defaults to `false`.
 * Default: `false`
 * Use partial appends in the transmuxer and segment loader
 
+##### liveRangeSafeTimeDelta
+* Type: `boolean`,
+* Default: `false`
```
Comment (gkatsev) [inline README.md:457]: This is a number that defaults to the SAFE_TIME_DELTA

**Review by gkatsev — COMMENTED** [review]
Thanks!

File: src/playback-watcher.js (hunk 1 above)
Comment (dario-fiore) [reply src/playback-watcher.js]: Done 👍🏻

File: README.md (hunk 2 above)
Comment (dario-fiore) [reply README.md]: Done 👍🏻

**Comment by dario-fiore** [comment]
@gkatsev It seems that tests fail for timeout. I tried locally on 2.3.0 and errors are the same. Any ideas?

**Comment by gkatsev** [comment]
Not sure, I'll take a look.

File: src/playback-watcher.js (hunk 1 above)
Comment (dario-fiore) [reply src/playback-watcher.js]: @gkatsev Are the new changes ok?

Comment (gkatsev) [reply src/playback-watcher.js]: yes, I think so. We'll re-review and take it for a quick spin but likely good to go.

**Review by gkatsev — APPROVED** [review]

**Pull request merged**